app/src/main/java/it/niedermann/owncloud/notes/persistence/AbstractNotesDatabase.java

```diff
@@ -108,10 +115,19 @@ private void createAccountTable(@NonNull SQLiteDatabase db) {
                 key_modified + " INTEGER, " +
                 key_api_version + " TEXT, " +
                 key_color + " VARCHAR(6) NOT NULL DEFAULT '000000', " +
-                key_text_color + " VARCHAR(6) NOT NULL DEFAULT '0082C9')");
+                key_text_color + " VARCHAR(6) NOT NULL DEFAULT '0082C9', " +
+                "FOREIGN KEY(" + key_id + ") REFERENCES " + table_category + "(" + key_category_account_id + "));");
         createAccountIndexes(db);
     }
 
+    private void createCategoryTable(@NonNull SQLiteDatabase db) {
+        db.execSQL("CREATE TABLE " + table_category + "(" +
+                key_category_id + " INTEGER PRIMARY KEY AUTOINCREMENT, " +
+                key_category_account_id + " INTEGER, " +
+                key_category_title + " TEXT )");
```

Comment:
Do you guys think a combined `UNIQUE` constraint over accountId and title would make sense?

@korelstar @Isaac-Graham

Reply:
It is useless but it also can be done. Each time I want to insert a new row into the category. I will check whether there is such a row whose title is the same as what I want to insert (See getCategoryIdByTitle). If there is such a row, I will not insert a row but return the category ID I search. Otherwise, I will insert a new row and return the category ID inserted just now. So that  I think adding a `UNIQUE` contraint is useless.

Reply:
Yeah, the current implementation of the business level will prevent it, but i think the db should catch as many pitfalls as possible. What can be done at database level should be done there, in this case additionally i think.

It will help us to keep a clean state even if we refactor the business lebel

Reply:
I agree with @stefan-niedermann